velox/vector/VectorStream.cpp

```diff
@@ -133,4 +133,44 @@ void VectorStreamGroup::read(
   getVectorSerde()->deserialize(source, pool, type, result, options);
 }
 
+folly::IOBuf rowVectorToIOBuf(
+    const RowVectorPtr rowVector,
+    memory::MemoryPool& pool) {
+  return rowVectorToIOBuf(rowVector, rowVector->size(), pool);
+}
+
+folly::IOBuf rowVectorToIOBuf(
+    const RowVectorPtr rowVector,
+    vector_size_t rangeEnd,
+    memory::MemoryPool& pool) {
+  auto streamGroup = std::make_unique<VectorStreamGroup>(&pool);
+  streamGroup->createStreamTree(asRowType(rowVector->type()), rangeEnd);
+
+  IndexRange range{0, rangeEnd};
+  streamGroup->append(rowVector, folly::Range<IndexRange*>(&range, 1));
+
+  IOBufOutputStream stream(pool);
+  streamGroup->flush(&stream);
+  return std::move(*stream.getIOBuf());
+}
+
+RowVectorPtr IOBufToRowVector(
+    const folly::IOBuf& ioBuf,
+    const RowTypePtr& outputType,
+    memory::MemoryPool& pool) {
+  std::vector<ByteRange> ranges;
+  ranges.reserve(4);
```

Comment:
why 4?

Reply:
@mbasmanova  I'm happy to change it in here, but I just wanted to have a small reserved buffer to prevent it from reallocating too often. The issue is that getting the actual size (number of buffers) is an IOBuf is linear on the linked list, so likely pretty slow. But unclear at this point is this matters or not